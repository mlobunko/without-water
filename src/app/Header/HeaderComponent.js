import React from 'react';
import { LinkContainer } from 'react-router-bootstrap';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';

export const HeaderComponent = ({
  authorized,
  authorizedActionsLogOut,
  isOpen,
  toggle
}) => (
  <Navbar color="dark" dark expand="md">
    <LinkContainer to="/">
      <NavbarBrand className="mr-auto">
        <div className="nav-bar__logo" />
      </NavbarBrand>
    </LinkContainer>
    <NavbarToggler onClick={toggle} className="mr-2" />
    <Collapse isOpen={isOpen} navbar>
      <Nav className="ml-auto" navbar>
        <LinkContainer to="/news" activeClassName="active">
          <NavItem>
            <NavLink className="nav-bar__link">News</NavLink>
          </NavItem>
        </LinkContainer>
        <LinkContainer to="/profile" activeClassName="active">
          <NavItem>
            <NavLink className="nav-bar__link">Profile</NavLink>
          </NavItem>
        </LinkContainer>
        {authorized ? (
          <LinkContainer exact to="/">
            <NavItem onClick={authorizedActionsLogOut}>
              <NavLink className="nav-bar__link">Log out</NavLink>
            </NavItem>
          </LinkContainer>
        ) : (
          <LinkContainer to="/login" activeClassName="active">
            <NavItem>
              <NavLink className="nav-bar__link">Log in</NavLink>
            </NavItem>
          </LinkContainer>
        )}
      </Nav>
    </Collapse>
  </Navbar>
);
