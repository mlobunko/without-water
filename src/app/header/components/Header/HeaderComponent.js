import React from 'react';
import { LinkContainer } from 'react-router-bootstrap';
import { Collapse, Navbar, NavbarToggler, Nav, NavItem } from 'reactstrap';
import PropTypes from 'prop-types';

import { StyledNavbarBrand, StyledNavLink } from './styles';

export const HeaderComponent = ({
  authorized,
  authorizedActionsLogOut,
  isOpen,
  toggle
}) => (
  <Navbar color="dark" dark expand="md">
    <LinkContainer to="/">
      <StyledNavbarBrand />
    </LinkContainer>
    <NavbarToggler onClick={toggle} className="mr-2" />
    <Collapse isOpen={isOpen} navbar>
      <Nav className="ml-auto" navbar>
        <LinkContainer to="/news" activeClassName="active">
          <NavItem>
            <StyledNavLink>News</StyledNavLink>
          </NavItem>
        </LinkContainer>
        <LinkContainer to="/profile" activeClassName="active">
          <NavItem>
            <StyledNavLink>Profile</StyledNavLink>
          </NavItem>
        </LinkContainer>
        {authorized ? (
          <LinkContainer exact to="/">
            <NavItem onClick={authorizedActionsLogOut}>
              <StyledNavLink>Log out</StyledNavLink>
            </NavItem>
          </LinkContainer>
        ) : (
          <LinkContainer to="/login" activeClassName="active">
            <NavItem>
              <StyledNavLink>Log in</StyledNavLink>
            </NavItem>
          </LinkContainer>
        )}
      </Nav>
    </Collapse>
  </Navbar>
);

export default HeaderComponent;

HeaderComponent.propTypes = {
  authorized: PropTypes.bool.isRequired,
  authorizedActionsLogOut: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired
};
