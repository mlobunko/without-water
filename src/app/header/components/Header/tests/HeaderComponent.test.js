import React from 'react';
import { shallow } from 'enzyme';

import { HeaderComponent } from '../HeaderComponent';

const initialProps = {
  authorized: true,
  authorizedActionsLogOut: jest.fn(),
  isOpen: true,
  toggle: jest.fn()
};

describe('render', () => {
  it('renders correctly if authorized true', () => {
    const props = {
      ...initialProps
    };
    const wrapper = shallow(<HeaderComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly if authorized false', () => {
    const props = {
      ...initialProps,
      authorized: false
    };
    const wrapper = shallow(<HeaderComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('logOut link', () => {
  const props = {
    ...initialProps
  };
  it('click', () => {
    const wrapper = shallow(<HeaderComponent {...props} />);
    wrapper.find({ onClick: props.authorizedActionsLogOut }).simulate('click');
    expect(props.authorizedActionsLogOut).toHaveBeenCalledTimes(1);
  });
});
