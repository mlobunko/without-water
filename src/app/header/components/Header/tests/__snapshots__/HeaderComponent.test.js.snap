// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly if authorized false 1`] = `
<Navbar
  color="dark"
  dark={true}
  expand="md"
  tag="nav"
>
  <LinkContainer
    activeClassName="active"
    aria-label="Home"
    exact={false}
    replace={false}
    strict={false}
    to="/without-water/"
  >
    <Styled(NavbarBrand) />
  </LinkContainer>
  <NavbarToggler
    aria-label="toggle drop menu"
    className="mr-2"
    onClick={[Function]}
    tag="button"
    type="button"
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={true}
    mountOnEnter={false}
    navbar={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/without-water/news"
      >
        <NavItem
          tag="li"
        >
          <Styled(NavLink)>
            News
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/without-water/profile"
      >
        <NavItem
          tag="li"
        >
          <Styled(NavLink)>
            Profile
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/without-water/login"
      >
        <NavItem
          tag="li"
        >
          <Styled(NavLink)>
            Log in
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
    </Nav>
  </Collapse>
</Navbar>
`;

exports[`render renders correctly if authorized true 1`] = `
<Navbar
  color="dark"
  dark={true}
  expand="md"
  tag="nav"
>
  <LinkContainer
    activeClassName="active"
    aria-label="Home"
    exact={false}
    replace={false}
    strict={false}
    to="/without-water/"
  >
    <Styled(NavbarBrand) />
  </LinkContainer>
  <NavbarToggler
    aria-label="toggle drop menu"
    className="mr-2"
    onClick={[Function]}
    tag="button"
    type="button"
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={true}
    mountOnEnter={false}
    navbar={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/without-water/news"
      >
        <NavItem
          tag="li"
        >
          <Styled(NavLink)>
            News
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/without-water/profile"
      >
        <NavItem
          tag="li"
        >
          <Styled(NavLink)>
            Profile
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={true}
        replace={false}
        strict={false}
        to="/without-water/"
      >
        <NavItem
          onClick={[Function]}
          tag="li"
        >
          <Styled(NavLink)>
            Log out
          </Styled(NavLink)>
        </NavItem>
      </LinkContainer>
    </Nav>
  </Collapse>
</Navbar>
`;
